runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -272,7 +272,14 @@ public TerminalNode getToken(int ttype, int i) {
 
 	@Override
 	public Interval getSourceInterval() {
-		if ( start==null || stop==null ) return Interval.INVALID;
+		if ( stop==null ) {
```

Comment:
:exclamation: This method only works when `start` is not null. The first thing in the method should be:

```java
if (start == null) {
  return Interval.INVALID;
}
```